### DISCUSSION TITLE
Issues with OSC 133 scrolling the buffer on keypress (nushell)

### DISCUSSION THREAD

**Discussion by SpacewaIker** [wezterm/wezterm#5859] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Nushell has support for shell integrations through some OSC codes, as indicated in the [default config](https://github.com/nushell/nushell/blob/e2d0514bb57ae0b071631d880acbcb695858f67c/crates/nu-utils/src/sample_config/default_config.nu#L236C5-L263C6).

However, it seems that the OSC 133 poses issues with Wezterm: when I have it enabled, typing in the terminal will scroll the text above. That is, whatever is printed above the current prompt will go up by one line for each character typed. If I set `osc133: false` in the nushell config, then this bug is resolved. And this doesn't happen in other terminal emulators such as Windows Terminal.

I wonder if this is an issue with my Wezterm configuration? Or perhaps it could be a bug with either nushell or wezterm?

Thanks!

**Comment by JustBarnt** [comment]
You are not the only one, and I have been bashing my head against my desk trying to figure out what was causing this, so I am so glad I found this. I too can set `osc133: false` in my nushell config and then it resolves the issues described above as well.

I'm so glad I found this, I love wezterm but started using nushell about 3 months ago and for the life of me could not figure out what was causing this, so I have been stuck using WindowsTerminal!

**Comment by meshal-h** [comment]
I faced the same issue, and setting `osc133: false` in the nushell config resolved it.

**Comment by zjm54321** [comment]
Thanks for this answer. Slove my problems.

**Comment by foxmean** [comment]
Just to add a verbatim command here because I cannot found a source how exactly to do via DuckDuckGo search but found accidentally by interacting with nushell repl itself.

So basically you add

`$env.config.shell_integration.osc133 = false`

to your nushell config file

**Comment by ask4fusora** [comment]
Guys, should we turn this into an issue?

**Comment by LukasBauza** [comment]
For anyone that wants this to only be done when using WezTerm, you can use this to be more explicit:

```nu
if ($env.TERM_PROGRAM == "WezTerm") {
  $env.config.shell_integration.osc133 = false
}
```
